### PR TITLE
feat: Cleanup AbstractModel

### PR DESCRIPTION
**Pull request by samuel-hawker**
Cleanup includes, argument renaming, documenting some methods
organizing fields into more logical groupings

Signed-off-by: Samuel Hawker <samuel.hawker@ibm.com>

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 1]
```diff
@@ -88,40 +88,50 @@
...
+    public static final String ANNO_STRIMZI_IO_DELETE_CLAIM = Annotations.STRIMZI_DOMAIN + "delete-claim";
+    public static final String ANNO_STRIMZI_LOGGING_HASH = Annotations.STRIMZI_DOMAIN + "logging-hash";
+
     @Deprecated
     public static final String ANNO_CO_STRIMZI_IO_DELETE_CLAIM = ClusterOperator.STRIMZI_CLUSTER_OPERATOR_DOMAIN + "/delete-claim";
 
-    public static final String ANNO_STRIMZI_CM_GENERATION = Annotations.STRIMZI_DOMAIN + "cm-generation";
-    public static final String ANNO_STRIMZI_LOGGING_HASH = Annotations.STRIMZI_DOMAIN + "logging-hash";
-
+    private static final String ENV_VAR_HTTP_PROXY = "HTTP_PROXY";
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:114]: Do you think this is really needed? I originally had it. But it seemed wasteful to have 3 static Strings which are used only in the static initilizer below. So I decided to remove them and use the env var names directly.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 2]
```diff
@@ -518,57 +556,59 @@ protected void setConfiguration(AbstractConfiguration configuration) {
...
+     * @param podId The Id (ordinal) of the pod.
      * @return The name of the pod with the given name.
      */
     public String getPodName(int podId) {
         return name + "-" + podId;
     }
 
     /**
-     * Sets the affinity as configured by the user in the cluster CR
+     * Sets the affinity as configured by the user in the cluster CR.
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:587]: Should we put empty line between the sentence and the @param? TBH I do not really mind, but it seems you did that in other Javadoc comments around.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 3]
```diff
@@ -638,17 +693,22 @@ protected PersistentVolumeClaim createPersistentVolumeClaim(int podNumber, Strin
 
         String storageClass = storage.getStorageClass();
         if (storage.getOverrides() != null) {
-            storageClass = storage.getOverrides().stream().filter(broker -> broker != null && broker.getBroker() != null && broker.getBroker() == podNumber && broker.getStorageClass() != null)
+            storageClass = storage.getOverrides().stream()
+                    .filter(broker -> broker != null &&
+                            broker.getBroker() != null &&
+                            broker.getBroker() == ordinalId &&
+                            broker.getStorageClass() != null)
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:700]: I do not think there is a real rule for this or anything like that, so feel free to leave it like this ... But I usually prefer to have the `&&` at the next line - it seems to me more readable.

```suggestion
                    .filter(broker -> broker != null 
                            && broker.getBroker() != null 
                            && broker.getBroker() == ordinalId 
                            && broker.getStorageClass() != null)
```

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java (hunk 1 above)
Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:114]: I know what you mean, I think the fact we use the same strings more than once means really we should use static strings.

Ultimately it all gets compiled down the same (correct me if I am wrong) and the consistency of always defining envars as static strings can only increase consistency and less bugs in the long run..

Not a hill I will die on for this particular envar, but still...

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java (hunk 3 above)
Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:700]: Great suggestion!

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 4]
```diff
@@ -349,12 +373,13 @@ public static OrderedProperties getOrderedProperties(String configFileName) {
...
-     * @param properties map with properties
-     * @return
+     * Transforms map to log4j properties file format.
+     *
+     * @param properties map of log4j properties.
+     * @return log4j properties as a String.
      */
     protected static String createPropertiesString(OrderedProperties properties) {
-        return properties.asPairsWithComment("Do not change this generated file. Logging can be configured in the corresponding kubernetes/openshift resource.");
+        return properties.asPairsWithComment("Do not change this generated file. Logging can be configured in the corresponding Kubernetes/OpenShift resource.");
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:382]: I would leave just Kubernetes.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaBridgeAssemblyOperatorTest.java [hunk 5]
```diff
@@ -80,7 +80,7 @@
     protected static Vertx vertx;
     private static final String METRICS_CONFIG = "{\"foo\":\"bar\"}";
     private static final String LOGGING_CONFIG = AbstractModel.getOrderedProperties("kafkaBridgeDefaultLoggingProperties")
-            .asPairsWithComment("Do not change this generated file. Logging can be configured in the corresponding kubernetes/openshift resource.");
+            .asPairsWithComment("Do not change this generated file. Logging can be configured in the corresponding Kubernetes/OpenShift resource.");
```
Comment (ppatierno) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaBridgeAssemblyOperatorTest.java:83]: As before I would say just Kubernetes. The same for the others following this one.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java (hunk 4 above)
Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:382]: As in remove the reference to OpenShift all together?

Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:382]: I would say that where the code is not related to OpenShift specific stuff, we should just mention Kubernetes.

Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:382]: Can I do this in a followup PR Paolo?

Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: Addressed :)

**Pull request merged**